activerecord/lib/active_record/connection_adapters/postgresql/schema_definitions.rb

```diff
@@ -124,11 +30,49 @@ class TableDefinition < ActiveRecord::ConnectionAdapters::TableDefinition
         # require you to assure that you always provide a UUID value before saving
         # a record (as primary keys cannot be +nil+). This might be done via the
         # +SecureRandom.uuid+ method and a +before_save+ callback, for instance.
-        def primary_key(name, type = :primary_key, options = {})
+        def primary_key(name, type = :primary_key, **options)
           options[:default] = options.fetch(:default, 'uuid_generate_v4()') if type == :uuid
           super
         end
 
+        [
+          :bigserial,
+          :bit,
+          :bit_varying,
+          :cidr,
+          :citext,
+          :daterange,
+          :hstore,
+          :inet,
+          :int4range,
+          :int8range,
+          :json,
+          :jsonb,
+          :ltree,
+          :macaddr,
+          :money,
+          :numrange,
+          :point,
+          :serial,
+          :tsrange,
+          :tstzrange,
+          :tsvector,
+          :uuid,
+          :xml,
+        ].each do |column_type|
+          define_method column_type do |*args, **options|
```

Comment:
Methods defined using `define_method` are slightly slower to execute than those defined by `def` so not recommended for hotspots. Similarly, splatting and globbing *args for most methods that only have one (all but xml and tsvector here) could be a performance penalty.

Perhaps make a `define_column_type` factory method that goes something like `define_column_for('something', args: true)` (or `false`) which would generate the appropriate method using the pattern:

```ruby
eval <<-EOS
  def(...)
    ...
  end
end
```